src/components/material.js

```diff
@@ -176,20 +177,40 @@ module.exports.Component = registerComponent('material', {
    */
   updateTexture: function (src) {
     var data = this.data;
+    var self = this;
     var material = this.material;
     if (src) {
       if (src !== this.textureSrc) {
         // Texture added or changed.
         this.textureSrc = src;
         srcLoader.validateSrc(src, loadImage, loadVideo);
+        // Material is first to load this texture. Load and resolve texture.
       }
     } else {
       // Texture removed.
       material.map = null;
       material.needsUpdate = true;
     }
-    function loadImage (src) { loadImageTexture(material, src, data.repeat); }
-    function loadVideo (src) { loadVideoTexture(material, src, data.width, data.height); }
+
```

Comment:
The logic to handle textures has moved to the texture module. It should be pretty much copying and pasting the code into https://github.com/aframevr/aframe/blob/dev/src/utils/texture.js